lib/Basics/StringUtils.cpp

```diff
@@ -1406,6 +1414,119 @@ std::string soundex(char const* src, size_t const len) {
   return result;
 }
     
+    
+template<typename InputType>
+inline bool eq(InputType const& c1, InputType const& c2) {
+    return c1 == c2;
+}
+
+template<typename InputType, typename LengthType>
+LengthType impl(InputType const* lhs,
+                               InputType const* rhs,
+                               LengthType lhsSize,
+                               LengthType rhsSize) {
+    assert(lhsSize >= rhsSize);
+        
+    std::vector<LengthType> costs;
+    costs.resize(rhsSize + 1);
+        
+    for (LengthType i = 0; i < rhsSize; ++i) {
+        costs[i] = i;
+    }
+        
+    LengthType next = 0;
+        
+    for (LengthType i = 0; i < lhsSize; ++i) {
+        LengthType current = i + 1;
+            
+        for (LengthType j = 0; j < rhsSize; ++j) {
+            LengthType cost = !(eq<InputType>(lhs[i], rhs[j]) ||
+                                (i && j && eq<InputType>(lhs[i - 1], rhs[j]) && eq<InputType>(lhs[i], rhs[j - 1])));
+            next = std::min(std::min(costs[j + 1] + 1, current + 1), costs[j] + cost);
+            costs[j] = current;
+            current = next;
+        }
+        costs[rhsSize] = next;
+    }
+    return next;
+}
+    
+size_t levenshteinDistance(std::vector<uint32_t> vect1, std::vector<uint32_t> vect2){
```

Comment:
vectors should be passed by const reference, not by value - otherwise they will be copied here.